Feature/loc 510 privacy manager

andrew-tran-infillion opened the pull request:
Add Android PrivacyManager module.

Add PrivacyManagerSpec for oldarch and newarch.
Add PrivacyManagerModule.
Add PrivacyManagerModule to GimbalPackage.
Fix Sample app _logConsentRequirement method to specify indeterminate requirement.

LeatherGreenGargoyle committed:
Add iOS PrivacyManager module.

Add javascript specs for PrivacyManager.
Add Sample app test code for PrivacyManager module.
Remove iOS Sample app API key.

LeatherGreenGargoyle committed:
Add Android PrivacyManager module.

Add PrivacyManagerSpec for android oldarch and newarch.
Add PrivacyManagerModule and add it to GimbalPackage.
Fix Sample app 'logConsentRequirement' method to specify indeterminate requirement.

LeatherGreenGargoyle committed:
Fix iOS PrivacyManager type issue and improve constants API.

Fix iOS PrivacyManager module method signatures to use doubles.
Add missing iOS PrivacyManager method getTurboModule.
Update Sample app Gimbal version.
Add PrivacyManager specs for constants.
Update Sample app to use new PrivacyManager constants API.

kimvsparrow commented on ios/RtnPrivacyManager.mm:
```diff
@@ -0,0 +1,112 @@
+
+#endif
+
+RCT_REMAP_METHOD(getGdprConsentRequirement, getGdprConsentRequirementWithResolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
```
These three aren't compatible with the codegen signatures:
```
- (void)getGdprConsentRequirement:(RCTPromiseResolveBlock)resolve
                           reject:(RCTPromiseRejectBlock)reject;
- (void)setUserConsent:(double)consentType
                 state:(double)state;
- (void)getUserConsent:(double)consentType
               resolve:(RCTPromiseResolveBlock)resolve
                reject:(RCTPromiseRejectBlock)reject;
```
The first one needs to be changed to `RCT_EXPORT_METHOD` and then `getGdprConsentRequirementWithResolver` can be snipped out to leave the resolve. I think for the others, the message names need to match (`reject` instead of `rejecter` etc.)

kimvsparrow commented on ios/RtnPrivacyManager.mm:
```diff
@@ -0,0 +1,112 @@
+
+@end
+
+@implementation RCTConvert(GMBLConsentType)
```
Apparently codegen doesn't support enums in signatures (yet, see https://github.com/reactwg/react-native-new-architecture/discussions/68#discussioncomment-3413488). So I don't think these are used any more.  It might take some manually testing to be sure though.

kimvsparrow commented on src/NativePrivacyManager.ts:
```diff
@@ -0,0 +1,37 @@
+    getUserConsent(consentType: Int32): Promise<Int32>;
+
+    readonly getConstants: () => {
+        CONSENT_STATE_UNKNOWN: Int32,
```
If I understand this declaration correctly, we're basically saying that `getConstants` is a lambda/closure that returns a type which is defined by all the fields in the curly brackets. Each of those typed fields should be terminated with a `;` (semi colon).

kimvsparrow commented on android/src/oldarch/com/gimbal/rtn/PrivacyManagerSpec.java:
```diff
@@ -0,0 +1,30 @@
+
+    public abstract void getGdprConsentRequirement(Promise promise);
+
+    public abstract void setUserConsent(int consentType, int state);
```
The codegen signatures for these two are:
```
  public abstract void setUserConsent(double consentType, double state);
  public abstract void getUserConsent(double consentType, Promise promise);
```

Which hopefully plays okay with a simple cast in the implementations.

andrew-tran-infillion replied:
Android testing looks good so far, but I'll keep this in mind if any weird bugs come up later.

kimvsparrow reviewed (changes requested):
Almost there!

LeatherGreenGargoyle committed:
fix: privacy manager method signatures and native spec

Fix PrivacyManager method signature mismatches.
Remove unused PrivacyManager enum converters.
Rename sample app test metho _logConsentRequirement to _logGDPRConsentRequirement.
Fix NativePlaceManager.getConstants syntax.

LeatherGreenGargoyle committed:
fix: android privacy manager method signature typing

LeatherGreenGargoyle committed:
style: add eslint dependencies and fix various formatting issues

Add eslint dependencies.
Add eslintrc to gitignore.
Fix App.tsx formatting issues.
Fix index.tsx formatting issues.
Fix NativePrivacyManager formatting issues.

kimvsparrow committed:
chore: revert eslint changes

Remove eslint dependencies from sample package.  Allow prettier to "fix"
its various style errors, now that eslint can't conflict with it.

LeatherGreenGargoyle committed:
fix: droid privacy manager spec method signature typing

LeatherGreenGargoyle committed:
fix: gradle sync issue

LeatherGreenGargoyle committed:
chore: update changelog to include privacy manager module

LeatherGreenGargoyle committed:
chore: remove unnecessary .gitignore entry

kimvsparrow reviewed (approved):
What a rush!
